### PR TITLE
Display favorite buttons with codelabs

### LINKED ISSUE #164: Display favorite buttons with codelabs and change so that they can be favorite
## Overview (Required)
-
image

- We can use session#sessionType

```
  enum class SessionType {
        @SerializedName("normal")
        Normal,
        @SerializedName("welcome_talk")
        WelcomeTalk,
        @SerializedName("reserved")
        Reserved,
        @SerializedName("codelabs")
        Codelabs,
        @SerializedName("fireside_chat")
        FiresideChat,
        @SerializedName("lunch")
        Lunch,
        @SerializedName("break")
        Break,
        @SerializedName("after_party")
        AfterParty
    }
```

## Links
- https://droidkaigi2019-dev.appspot.com/api/timetable

### PR DESCRIPTION
**Pull request by kirimin**
## Issue
- close #164

## Overview (Required)
- Display favorite buttons with codelabs and change so that they can be favorite.

## Links
-

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
Thanks! Almost LGTM!
I think this is a little hard issue 👏

**Comment by takahirom** [comment]
Can you add `isFavoritable` field to SessionType?

File: model/src/commonMain/kotlin/Session.kt [hunk 1]
```diff
@@ -40,8 +41,9 @@ sealed class Session(
         override val endTime: DateTime,
         val title: String,
         override val room: Room,
-        val sessionType: SessionType
-    ) : Session(id, dayNumber, startTime, endTime, room)
+        val sessionType: SessionType,
+        override val isFavorited: Boolean = true
```
Comment (takahirom) [inline model/src/commonMain/kotlin/Session.kt]: Is default argument needed? 🔖

**Comment by kirimin** [comment]
I got it!
I'll add it as soon as.

File: model/src/commonMain/kotlin/Session.kt (hunk 1 above)
Comment (kirimin) [reply model/src/commonMain/kotlin/Session.kt]: Nooo!
It's debug code... 😢

File: feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/item/ServiceSessionItem.kt [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+import com.squareup.inject.assisted.AssistedInject
 import com.xwray.groupie.databinding.BindableItem
 import io.github.droidkaigi.confsched2019.model.Session
 import io.github.droidkaigi.confsched2019.session.R
 import io.github.droidkaigi.confsched2019.session.databinding.ItemServiceSessionBinding
+import io.github.droidkaigi.confsched2019.session.ui.actioncreator.SessionContentsActionCreator
 
-class ServiceSessionItem(
-    override val session: Session.ServiceSession
+class ServiceSessionItem @AssistedInject constructor(
```
Comment (takahirom) [inline feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/item/ServiceSessionItem.kt]: 💯

File: model/src/commonMain/kotlin/Session.kt (hunk 1 above)
Comment (kirimin) [reply model/src/commonMain/kotlin/Session.kt]: I removed it!

File: feature/session/src/main/res/layout/item_service_session.xml [hunk 3]
```diff
@@ -73,7 +73,7 @@
             android:scaleType="centerInside"
             android:src="@{session.isFavorited ? @drawable/ic_bookmark_black_24dp : @drawable/ic_bookmark_border_black_24dp}"
             android:tint="@{session.isFavorited ? @color/colorPrimary : @color/gray1}"
-            android:visibility="@{session.sessionType == SessionType.Codelabs ? View.VISIBLE : View.GONE}"
+            android:visibility="@{session.sessionType.isFavoritable ? View.VISIBLE : View.GONE}"
```
Comment (takahirom) [inline feature/session/src/main/res/layout/item_service_session.xml]: You can use `app:visibleGone` attribute 😉

**Comment by kirimin** [comment]
I added ```isFavoritable``` and use it for is favorite button visible.

**Comment by kirimin** [comment]
I'll fix the CI

**Comment by takahirom** [comment]
🙏

**Comment by kirimin** [comment]
I may fixed CI.
https://circleci.com/gh/DroidKaigi/conference-app-2019/tree/pull%2F352

**Comment by takahirom** [comment]
LGTM 🔖

**Comment by kirimin** [comment]
Thank you for reviewing and great advices until late at night !! :pray:

**Pull request merged**